### PR TITLE
Update copyright header from 2022 to 2023

### PR DESCRIPTION
**Pull request by wanyufe**
*Issue #, if available:*
Copyright header **2022** causing build warning.
*Description of changes:*
Update *.go copyright header
*Testing performed:*
Unit testing

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: api/v1beta1/cloudstackaffinitygroup_conversion.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2022 The Kubernetes Authors.
+Copyright 2023 The Kubernetes Authors.
```
Comment (mrog) [inline api/v1beta1/cloudstackaffinitygroup_conversion.go:2]: Would it be better to change it to `2022-2023`?

Comment (wanyufe) [reply api/v1beta1/cloudstackaffinitygroup_conversion.go:2]: I honest don't know. This maybe a legal question. I did not see 2021-2022 in vsphere .go files even we know it started before 2022.

Comment (mrog) [reply api/v1beta1/cloudstackaffinitygroup_conversion.go:2]: I looked at some of the vsphere files, and they contain the year that the file was created, not the year it was last updated.  Maybe we should leave the copyrights alone?

Comment (wanyufe) [reply api/v1beta1/cloudstackaffinitygroup_conversion.go:2]: I agree with you on this. 
But our repo has a `goheader` check that tries to make sure the copyright year is current year. Maybe we should not include it in our lint checking list. 

@davidjumani @jweite-amazon Do you have any opinion on this?

Comment (davidjumani) [reply api/v1beta1/cloudstackaffinitygroup_conversion.go:2]: Looking at other projects, it is the year created, not the current year. IMO the linter should be changed

**Comment by wanyufe** [comment]
close, create a new solution: https://github.com/kubernetes-sigs/cluster-api-provider-cloudstack/pull/211

**Pull request closed without merging**